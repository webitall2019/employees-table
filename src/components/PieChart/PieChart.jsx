import React, { useState, useEffect, useMemo } from "react";
import { map, countBy } from "lodash";
import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";
import { Pie } from "react-chartjs-2";
import { useSelector } from "react-redux";

ChartJS.register(ArcElement, Tooltip, Legend);

const PieChart = () => {
  const users = useSelector((state) => state.users);
  const [devCount, setDevCount] = useState(0);
  const [qaCount, setQaCount] = useState(0);
  const [jobNames, setjobNames] = useState([])

  // const [filteredUsers, setFilteredUsers] = useState([]);

  const jobTitles = map(users, (obj) => obj.jobTitle);

  useEffect(() => {
    // setFilteredUsers(users);
    // setDevCount(countBy(jobTitles).Developer);
    
    setDevCount(jobTitles.filter((el) => el === "Developer").length);
    
    setQaCount(countBy(jobTitles).QA);
  }, [users, jobTitles]);

  const dataChart = useMemo(() => {
    const data = {
      labels: ["Developer", "QA"],
      datasets: [
        {
          data: [devCount, qaCount],
          backgroundColor: [
            "rgba(255, 99, 132, 0.2)",
            "rgba(54, 60, 235, 0.2)",
          ],
          borderColor: ["rgba(255, 99, 132, 1)", "rgba(54, 162, 235, 1)"],
          borderWidth: 1,
        },
      ],
    };
    return data;
  }, [devCount, qaCount]);

  return (
    <div className="pie-chart chart" data-testid="pieChart">
      <Pie data={dataChart} />
    </div>
  );
};
export default PieChart;
