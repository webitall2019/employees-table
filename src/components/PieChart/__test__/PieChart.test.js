import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import store from "../../../store/store";
import PieChart from "../PieChart";

describe("<PieChart/> component ", () => {
  it("check to be in the document", () => {
    render(
      <Provider store={store}>
        <PieChart />
      </Provider>
    );
    const pieChartComponent = screen.getByTestId("pieChart");
    expect(pieChartComponent).toBeInTheDocument();
    
  });
  it("should match snapshot WHEN rendered", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <PieChart />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
