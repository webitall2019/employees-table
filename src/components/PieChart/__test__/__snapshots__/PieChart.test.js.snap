// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieChart/> component  should match snapshot WHEN rendered 1`] = `
<DocumentFragment>
  <div
    class="pie-chart chart"
    data-testid="pieChart"
  />
</DocumentFragment>
`;
